lib/userMemoryStore.js

```diff
@@ -96,13 +98,37 @@ exports.updatePassword = function(username, oldPwd, newPwd, cb) {
   });
 };
 
+function calculateDelay(attempts) {
+  return (Math.pow(2, attempts) - 1) * ATTEMPT_DELAY_MS;
+}
+
+function doVerify(password, user, cb) {
+  // user.password here should be the stored hash
+  hash.verify(password, user.password, function(err, match) {
+    if (err) {
+      return cb(err);
+    }
+    if (!match) {
+      user.passwordAttempts = user.passwordAttempts || 0;
+      user.passwordAttempts++;
+    } else {
+      user.passwordAttempts = 0;
+    }
+    cb(null, match);
+  });
+}
+
 exports.verifyPassword = function(username, password, cb) {
   const user = findByUsername(username);
   if (!user) {
     return cb(new Error('User not found'));
```

Comment:
This module runs inside our infrastructure and is only accessible by the raincatcher-demo-cloud front end, and from there to the client apps, so I'd like to leave that decision to the client code.

Also adding more global logging to the raincatcher modules is in our backlog (only `console.log` atm), and we can look into this kind of audit trail logging too!